code/modules/paperwork/contract.dm

```diff
@@ -0,0 +1,306 @@
+
+
+/obj/item/weapon/paper/contract/employment/attack(mob/living/M, mob/living/carbon/human/user)
+	var/deconvert = 0
+	if(M.mind == target && target.soulOwner != target)
+		if(user.mind && (user.mind.assigned_role == "Internal Affairs Agent"))
+			deconvert = 25
+		else if (user.mind && (user.mind.assigned_role =="Head of Personnel") || (user.mind.assigned_role == "Centcom Commander") || (user.mind.assigned_role == "Magistrate"))
+			deconvert = 10 // the HoP doesn't have AS much legal training
+		else
+			deconvert = 0.0001 // one out of a million
+	if(prob(deconvert))
+		M.visible_message("<span class='notice'>[user] reminds [M] that [M]'s soul was already purchased by Nanotrasen!</span>")
+		to_chat(M,"<span class='boldnotice'>You feel that your soul has returned to its rightful owner, Nanotrasen.</span>")
+		M.return_soul()
+	return ..()
+
+
+/obj/item/weapon/paper/contract/infernal
+	var/contractType = 0
+	burn_state = LAVA_PROOF
+	var/datum/mind/owner
+	icon_state = "paper_onfire"
+
+/obj/item/weapon/paper/contract/infernal/power
+	name = "paper- contract for infernal power"
+	contractType = CONTRACT_POWER
+
+/obj/item/weapon/paper/contract/infernal/wealth
+	name = "paper- contract for unlimited wealth"
+	contractType = CONTRACT_WEALTH
+
+/obj/item/weapon/paper/contract/infernal/prestige
+	name = "paper- contract for prestige"
+	contractType = CONTRACT_PRESTIGE
+
+/obj/item/weapon/paper/contract/infernal/magic
+	name = "paper- contract for magical power"
+	contractType = CONTRACT_MAGIC
+
+/obj/item/weapon/paper/contract/infernal/revive
+	name = "paper- contract of resurrection"
+	contractType = CONTRACT_REVIVE
+
+/obj/item/weapon/paper/contract/infernal/knowledge
+	name = "paper- contract for knowledge"
+	contractType = CONTRACT_KNOWLEDGE
+
+/obj/item/weapon/paper/contract/infernal/unwilling
```

Comment:
You appear to have left out the rest of the `unwilling` stuff, but not this. Any reason?